typedb/connection/client.py

```diff
@@ -42,11 +43,13 @@ def __init__(self, address: str, parallelisation: int = 2):
         self._transmitter = RequestTransmitter(parallelisation)
         self._sessions: Dict[bytes, _TypeDBSessionImpl] = {}
         self._sessions_lock = Lock()
-        self._is_open = True
         self._pulse_executor = ScheduledExecutor()
         self._pulse_executor.schedule_at_fixed_rate(interval=self._PULSE_INTERVAL_SECONDS, action=self._transmit_pulses)
 
     def session(self, database: str, session_type: SessionType, options=None) -> _TypeDBSessionImpl:
+        if not self.is_open():
```

Comment:
since opening a client is now more intricate, we validate the client is fully opened before allowing sessions to be opened